[gitlab] Simplify `package_build` scripts

KSerrania opened the pull request:
### What does this PR do?

Simplifies the `package_build` job scripts:
- removes test install steps (covered & done better by kitchen tests)
- simplifies the logic to fetch omnibus artifacts and put them in a correct spot to be picked up by gitlab.

### Motivation

Requires https://github.com/DataDog/omnibus-ruby/pull/119.
Omnibus artifacts are already copied in the workspace (in `omnibus/pkg`) at the end of the build, so there's no need for the additional artifact copy logic.
On Linux, we were doing some tests on the package (listing contents for deb packages, package install for rpm paclages) that should already be covered by kitchen tests.
Adds some missing artifacts: on Windows, the `metadata.json` files, which contain the software definition lists w/ dependencies, and the `manifest-version.json` file on all platforms.

### Describe your test plan

Check that the artifacts of new pipelines contain all artifacts that were generated on old pipelines.

KSerrania committed:
Simplify Linux package build artifact extraction

KSerrania committed:
Use omnibus-ruby branch

KSerrania committed:
Simplify Windows package_build artifact extraction

KSerrania committed:
Fix SUSE scripts

KSerrania committed:
Remove installation step in build jobs

KSerrania committed:
Remove unused omnibus variables

KSerrania committed:
Fix typo in comments

KSerrania committed:
Use stable omnibus-ruby branch

KSerrania committed:
Merge remote-tracking branch 'origin/master' into kserrania/test-package-dir

mx-psi commented on .gitlab-ci.yml:
```diff
@@ -83,18 +83,13 @@ variables:
   # Directory in which we put the artifacts after the build
   # Must be in $CI_PROJECT_DIR
-  OMNIBUS_PACKAGE_DIR_SUSE: $CI_PROJECT_DIR/.omnibus/suse/pkg
-  OMNIBUS_BASE_DIR_WIN: c:\omni-base\$CI_RUNNER_ID
-  OMNIBUS_BASE_DIR_WIN_OMNIBUS: c:/omni-base/$CI_RUNNER_ID
+  OMNIBUS_PACKAGE_DIR_SUSE: $CI_PROJECT_DIR/omnibus/suse/pkg
```
Should we add a comment similar to the one with had before explaining the need for this?

KSerrania committed:
Improve comment about OMNIBUS_PACKAGE_DIR_SUSE

KSerrania commented:
It depends on what these jobs want to do. In the omnibus builds, we know that we want everything that ends up in the `omnibus/pkg` folder as an artifact, which is why we directly use it instead of using a temporary `build-out` folder

In the chocolatey package generation job, the script that generates the choco package targets `build-out`, and then only the `.nupkg` files from this folder [are copied to the final artifact folder](https://github.com/DataDog/datadog-agent/blob/3093c73f5105d250068bac1a6909943a10949fb0/.gitlab/choco_build.yml#L53). If there only are `nupkg` files in this folder to begin with, then it makes sense to skip this temporary folder. If not, and we don't want the other files, then it makes sense to keep it.

KSerrania committed:
Fix interaction with chocolatey offline job

KSerrania commented:
I pushed one last fix: this discussion reminded me that the offline chocolatey package generation uses the artifacts whose path I changed, so I fixed that.
